src/VirtoCommerce.Xapi.Data/Mapping/FacetMappingProfile.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Linq;
+using AutoMapper;
+using VirtoCommerce.SearchModule.Core.Model;
+using VirtoCommerce.Xapi.Core.Extensions;
+using VirtoCommerce.Xapi.Core.Models.Facets;
+
+namespace VirtoCommerce.Xapi.Data.Mapping;
+
+public class FacetMappingProfile : Profile
+{
+    public FacetMappingProfile()
+    {
+        CreateMap<Aggregation, FacetResult>().IncludeAllDerived().ConvertUsing((request, facet, context) =>
+        {
+            context.Items.TryGetValue("cultureName", out var cultureNameObj);
+            var cultureName = cultureNameObj as string;
+            FacetResult result = request.AggregationType switch
+            {
+                "attr" => new TermFacetResult
+                {
+                    Name = request.Field,
+                    Label = request.Field,
+                    Terms = request.Items?.Select(x => new FacetTerm
+                    {
+                        Count = x.Count,
+                        IsSelected = x.IsApplied,
+                        Term = x.Value?.ToString(),
+                        Label = x.Labels?.FirstBestMatchForLanguage(x => x.Language, cultureName)?.Label ?? x.Value?.ToString(),
+                    }).ToArray() ?? [],
+
+                },
+                "range" => new RangeFacetResult
+                {
+                    Name = request.Field,
+                    Label = request.Field,
+                    Ranges = request.Items?.Select(x => new FacetRange
+                    {
+                        Count = x.Count,
+                        IsSelected = x.IsApplied,
+                        From = Convert.ToInt64(x.RequestedLowerBound),
+                        IncludeFrom = x.IncludeLower,
+                        FromStr = x.RequestedLowerBound,
+                        To = Convert.ToInt64(x.RequestedUpperBound),
+                        IncludeTo = x.IncludeUpper,
+                        ToStr = x.RequestedUpperBound,
+                        Label = x.Value?.ToString(),
+                    }).ToArray() ?? [],
+                },
+                _ => null
+            };
+
+            return result;
```

Comment:
Add statistics and facet ordering. See  https://github.com/VirtoCommerce/vc-module-x-catalog/blob/9fdaf4caecc19a984d3f012bf13ed6548307a5f0/src/VirtoCommerce.XCatalog.Data/Mapping/FacetMappingProfile.cs#L51
and
 https://github.com/VirtoCommerce/vc-module-x-catalog/blob/9fdaf4caecc19a984d3f012bf13ed6548307a5f0/src/VirtoCommerce.XCatalog.Data/Mapping/FacetMappingProfile.cs#L67